src/OpenTelemetry.Resources.Container/ContainerResourceEventSource.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Diagnostics.Tracing;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Resources.Container;
+
+[EventSource(Name = "OpenTelemetry-Resources-Container")]
+internal class ContainerResourceEventSource : EventSource, IServerCertificateValidationEventSource
+{
+    public static ContainerResourceEventSource Log = new();
+
+    private const int EventIdFailedToExtractResourceAttributes = 1;
+    private const int EventIdFailedToValidateCertificate = 2;
+    private const int EventIdFailedToCreateHttpHandler = 3;
+    private const int EventIdFailedCertificateFileNotExists = 4;
+    private const int EventIdFailedToLoadCertificateInStorage = 5;
+
+    [NonEvent]
+    public void ExtractResourceAttributesException(string format, Exception ex)
+    {
+        if (this.IsEnabled(EventLevel.Error, (EventKeywords)(-1)))
+        {
+            this.FailedToExtractResourceAttributes(format, ex.ToInvariantString());
+        }
+    }
+
+    [Event(EventIdFailedToExtractResourceAttributes, Message = "Failed to extract resource attributes in '{0}'.", Level = EventLevel.Error)]
+    public void FailedToExtractResourceAttributes(string format, string exception)
+    {
+        this.WriteEvent(1, format, exception);
```

Comment:
Can EventIdFailedToExtractResourceAttributes be used here instead of hard-coded 1?

Reply:
Yes, done.